File: spring-web-java/src/main/java/spring/web/java/presentation/dto/response/RoleResponse.java
package spring.web.java.presentation.dto.response;

import java.util.List;

import spring.web.java.domain.model.entity.Permission;
import spring.web.java.domain.model.entity.Role;

public record RoleResponse(
	Long id,
	String name,
	List<PermissionResponse> permissions
) {
	public RoleResponse(Role role, List<Permission> permissions) {
		this(
			role.getId(),
			role.getName(),
			permissions.stream()
				.map(PermissionResponse::new)
				.toList()
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/response/RoleResponse.kt
package spring.web.kotlin.presentation.dto.response

import spring.web.kotlin.domain.model.entity.Permission
import spring.web.kotlin.domain.model.entity.Role

data class RoleResponse(
    val id: Long,
    val name: String,
    val permissions: List<PermissionResponse>
) {
    constructor(role: Role, permissions: List<Permission>) : this(
        id = role.id!!,
        name = role.name,
        permissions = permissions.map { PermissionResponse(it) }
    )
}